src/auth/src/credentials/api_key_credentials.rs

```diff
@@ -157,16 +157,12 @@ mod test {
                 metadata: None,
             }
         );
-        let headers: Vec<HV> = HV::from(creds.headers().await.unwrap());
+        let headers = creds.headers(None).await.unwrap();
+        let value = headers.get(API_KEY_HEADER_KEY).unwrap();
 
-        assert_eq!(
-            headers,
-            vec![HV {
-                header: API_KEY_HEADER_KEY.to_string(),
-                value: "test-api-key".to_string(),
-                is_sensitive: true,
-            }]
-        );
+        assert_eq!(headers.len(), 1);
```

Comment:
Here and below. When this fails you will want to know more:

```suggestion
        assert_eq!(headers.len(), 1, "{headers:?}");
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=85830927b4b6cb1289ef810e8c32bcca